src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -2366,7 +2383,27 @@ public Bundle performFhirSearch(Class<? extends Resource> resourceClass, Map<Str
 				query = query.where(new TokenClientParam(param).exactly().codes(values));
 			}
 		}
-		return query.returnBundle(Bundle.class).execute();
+
+		// Execute the initial search query
+		Bundle bundle = query.returnBundle(Bundle.class).execute();
+
+		// Check if there are more pages
+		while (bundle != null && bundle.getLink(Bundle.LINK_NEXT) != null) {
+			// Fetch the next page of results
+			String nextPageUrl = bundle.getLink(Bundle.LINK_NEXT).getUrl();
+			Bundle nextPageBundle = fhirClientAuthenticatorService.getFhirClient().search().byUrl(nextPageUrl).returnBundle(Bundle.class).execute();
+
+			// Add the resources from the next page to the current bundle
+			for (Bundle.BundleEntryComponent entry : nextPageBundle.getEntry()) {
+				bundle.addEntry(entry);
+			}
+
+			// Update the bundle to the next page bundle
+			bundle = nextPageBundle;
+		}
```

Comment:
Here my only concern is that the number of times this loop will run. We have to make sure that this completes before the search gets expired.

Reply:
It should not create any problem since we are already doing something similar in report gen and it is running in training & production without any issues.
  private suspend fun fetchBundleBySearchUrl(bundle: Bundle, url: String) {
    val searchBundle = fhirClient.search().byUrl(url).execute() as Bundle
    bundle.entry.addAll(searchBundle.entry)
    if (searchBundle.hasLink() && searchBundle.link.containsNext()) {
      fetchBundleBySearchUrl(bundle, searchBundle.link.getNextUrl())
    }
  }